src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const res = [];
+  let obj = Object.assign({}, state);
+
+  for (const action of actions) {
+    obj = Object.assign({}, obj, action.extraData);
```

Comment:
redundant. You'd better use object destructing
```suggestion
    const { type, extraData, keysToRemove } = action;
```